packages/zed-js/src/client/base-client.ts

```diff
@@ -52,22 +52,24 @@ export abstract class BaseClient {
 
   async describeQuery(
     query: string,
-    pool?: string,
-    options: { signal?: AbortSignal; timeout?: number } = {}
+    options: { signal?: AbortSignal; timeout?: number; pool?: string } = {}
   ) {
-    const head = pool ? { pool } : null;
+    const head = options.pool ? { pool: options.pool } : null;
     const abortCtl = wrapAbort(options.signal);
-    const result = await this.send({
+    const response = await this.request({
       method: 'POST',
       path: `/query/describe`,
       body: JSON.stringify({ query, head }),
       contentType: 'application/json',
       format: 'json',
       signal: abortCtl.signal,
       timeout: options.timeout,
-      dontRejectError: true,
     });
-    return result.json();
+    if (response.ok || response.status === 400) {
```

Comment:
Re-arranged the code to allow for an error response from the server. If the describe endpoint returns code 400, there is a problem with the query, but we don't want to throw an error. We want to report the problem to the user. So we parse the JSON as an expected response.